packages/flutter_web_plugins/lib/src/plugin_event_channel.dart

```diff
@@ -0,0 +1,93 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+import 'plugin_registry.dart';
+
+/// A named channel for sending events to the framework-side using streams.
+///
+/// This is the platform-side equivalent of [EventChannel]. Whereas
+/// [EventChannel] receives a stream of events from platform plugins, this
+/// channel sends a stream of events to the handler listening on the
+/// framework-side.
+class PluginEventChannel<T> {
+  /// Creates a new plugin event channel.
+  const PluginEventChannel(
```

Comment:
Document as an additional caveat that `[name]` must not be null, and `[codec]` if provided must not be null, defaults to `[StandardMethodCodec]`.

Reply:
Done.